src/Sulu/Component/SmartContent/Orm/DataProviderRepositoryTrait.php

```diff
@@ -58,21 +58,16 @@ private function findByFiltersIds($filters, $page, $pageSize, $limit, $locale, $
     {
         $parameter = [];
 
-        $queryBuilder = $this->createQueryBuilder('c')
-            ->select('c.id')
-            ->distinct()
-            ->orderBy('c.id', 'ASC');
+        $queryBuilder = $this->createQueryBuilder('c');
+
+        $parameter = array_merge($parameter, $this->appendSelect($queryBuilder, 'c', $locale, $options));
 
         $tagRelation = $this->appendTagsRelation($queryBuilder, 'c');
         $categoryRelation = $this->appendCategoriesRelation($queryBuilder, 'c');
 
-        if (array_key_exists('sortBy', $filters) && is_array($filters['sortBy'])) {
```

Comment:
`sortBy` should never be an array, right?

Reply:
This fix allows sorting during `findByFilterIds`

Reply:
i think it cannot be an array anmore. at least https://github.com/sulu/sulu/pull/5179/files#diff-3ba8d43dcc752376edcd2f5156c711bfL360 removed some functionality regarding to this too

Reply:
Right, it cannot be an array at all. Therefore you can remove the code handling sorting by multiple columns.